cmd/kops/create_cluster.go

```diff
@@ -549,22 +547,6 @@ func RunCreateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Cr
 		}
 	}
 
-	if c.Image != "" {
-		for _, group := range instanceGroups {
-			group.Spec.Image = c.Image
-		}
-	}
-	if c.MasterImage != "" {
-		for _, group := range masters {
-			group.Spec.Image = c.MasterImage
-		}
-	}
-	if c.NodeImage != "" {
-		for _, group := range nodes {
-			group.Spec.Image = c.NodeImage
-		}
-	}
-
```

Comment:
Where was this logic moved?

Reply:
E.g https://github.com/kubernetes/kops/pull/14127/files#diff-cf0e850db1e4461eab5d4dac860d20f2176b940f236e50825ae538ee23688a1eR861

So in `new_cluster.go`

Reply:
I don't see any reference to `NodeImage` and I don't see `Image` being set first and `MasterImage` and `NodeImage` overriding it.

Reply:
https://github.com/kubernetes/kops/pull/14127/files#diff-cf0e850db1e4461eab5d4dac860d20f2176b940f236e50825ae538ee23688a1eR396

It happens the other way around now. So if the image doesn't get set during IG creation, it gets set by the default logic.

But I think you are right about NodeImage not being respected. We may be missing a test case here. Let me try to add one.